Feature/refactor classes to hooks

oftheheadland opened the pull request:
## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3330906

## Main Changes:
* The last 3 class-based Community tabs (Restore, Identified Issues, and the Monitoring tab) are now refactored to use React Hooks. 

## Steps To Test:

Tabs to check:
**Monitoring
Restore
Identified Issues**

1. Navigate to a few testing locations on the Community page and try the above tabs to verify there are no regressions or new issues introduced.
2. Change locations while on the tab and check that new data populates correctly, toggle all switches, and check the Monitoring tab sorts.

Note that the Monitoring and Identified Issues tabs perform noticeably slower while Chrome devtools are opened. This issue has been in the app for a long time and can be reproduced on the dev and staging sites. As our users would not see this and have not complained so far I don't think it's an issue.

cschwinderg commented on app/client/src/components/pages/Community/components/tabs/Monitoring.js:
```diff
@@ -116,39 +107,45 @@ type StationGroups = {
+    watershed,
+  } = React.useContext(LocationSearchContext);
+
+  const [
+    prevMonitoringLocationData,
+    setPrevMonitoringLocationData,
+  ] = React.useState({});
```
I think it would be better to go ahead and include the `MonitoringLocationData` type (lines 94 - 101 in the old code). You can use it on the state like this

```suggestion
  const [
    prevMonitoringLocationData,
    setPrevMonitoringLocationData,
  ] = React.useState<MonitoringLocationData>({});
```

oftheheadland commented:
@cschwinderg  thanks for catching those issues! Tracked down some bugs around syncing the toggles to the map widget, added back in the MonitoringLocationData type, and fixed the issue where toggling a single parameter would cause it to not take effect.

oftheheadland commented:
@cschwinderg ran prettier again across the app again 👍

cschwinderg commented:
@oftheheadland Can you resolve the conflicts on this one?

oftheheadland commented:
@cschwinderg resolved conflicts
